build/lib/pyaadhaar/decode.py

```diff
@@ -69,49 +75,61 @@ def signedData(self):
 
     def isMobileNoRegistered(self):
         # Will return True if mobile number is registered
-        if int(self.data['email_mobile_status']) == 3 or int(self.data['email_mobile_status']) == 1:
-            return True
-        return False
+        return self.data['mobile']
 
     def isEmailRegistered(self):
         # Will return True if email id is registered
-        if int(self.data['email_mobile_status']) == 3 or int(self.data['email_mobile_status']) == 2:
-            return True
-        return False
+        return self.data['email']
 
     def sha256hashOfEMail(self):
         # Will return the hash of the email id
         tmp = ""
         if int(self.data['email_mobile_status']) == 3:
             tmp = self.decompressed_array[len(
                 self.decompressed_array)-256-32-32:len(self.decompressed_array)-256-32].hex()
-        elif int(self.data['email_mobile_status']) == 2:
+        elif int(self.data['email_mobile_status']) == 1:
             tmp = self.decompressed_array[len(
                 self.decompressed_array)-256-32:len(self.decompressed_array)-256].hex()
         return tmp
 
     def sha256hashOfMobileNumber(self):
         # Will return the hash of mobile number
         tmp = ""
-        if int(self.data['email_mobile_status']) == 3 or int(self.data['email_mobile_status']) == 1:
+        if int(self.data['email_mobile_status']) == 3 or int(self.data['email_mobile_status']) == 2:
             tmp = self.decompressed_array[len(
                 self.decompressed_array)-256-32:len(self.decompressed_array)-256].hex()
         return tmp
 
+    def isImage(self, buffer = 10) -> bool:
+        # Will return bool for availability of image stream in the QR CODE
+        if int(self.data['email_mobile_status']) == 3:
+            if len(self.decompressed_array[self.delimeter[len(self.details)]+1:len(self.decompressed_array)]) < 256+32+32+buffer:
+                return False
+            else:
+                return True
+        elif int(self.data['email_mobile_status']) == 2 or int(self.data['email_mobile_status']) == 1:
+            if len(self.decompressed_array[self.delimeter[len(self.details)]+1:len(self.decompressed_array)]) < 256+32+buffer:
+                return False
+            else:
+                return True
+        elif int(self.data['email_mobile_status']) == 0:
+    
     def image(self):
         # Will return the image stream to be used in another function
         if int(self.data['email_mobile_status']) == 3:
-            return Image.open(BytesIO(self.decompressed_array[self.delimeter[15]+1:len(self.decompressed_array)-256-32-32]))
+            print("length --> ", len(self.decompressed_array[self.delimeter[len(self.details)]+1:len(self.decompressed_array)]))
```

Comment:
remove the print statement, or add a debug value in class to print while debug is set to true

Reply:
Sure! print statement will be removed